src/xiaomi_ble/parser.py

```diff
@@ -1461,6 +1470,140 @@ def obj4e1c(
+    return {}
+
+
+def obj5414(
+    xobj: bytes, device: XiaomiBluetoothDeviceData, device_type: str
+) -> dict[str, Any]:
+    """Device mode (KSI and KSIBP, not used in HA)"""
+    return {"mode": xobj[0]}
+
+
+def obj560c(
+    xobj: bytes, device: XiaomiBluetoothDeviceData, device_type: str
+) -> dict[str, Any]:
+    """Button press"""
+    if device_type in ["KS1", "KS1BP"]:
+        # button click 1-4
+        button = xobj[0]
+        if button == 1:
+            # left button
+            device.fire_event(
+                key=f"{str(EventDeviceKeys.BUTTON)}_left",
+                event_type="press",
+                event_properties=None,
+            )
+        if button == 2:
+            # mid left button
+            device.fire_event(
+                key=f"{str(EventDeviceKeys.BUTTON)}_mid_left",
+                event_type="press",
+                event_properties=None,
+            )
+        if button == 3:
+            # mid right button
+            device.fire_event(
+                key=f"{str(EventDeviceKeys.BUTTON)}_mid_right",
+                event_type="press",
+                event_properties=None,
+            )
+        if button == 4:
+            # right button
+            device.fire_event(
+                key=f"{str(EventDeviceKeys.BUTTON)}_right",
+                event_type="press",
+                event_properties=None,
+            )
+        else:
+            return {}
+    return {}
+
+
+def obj560d(
+    xobj: bytes, device: XiaomiBluetoothDeviceData, device_type: str
+) -> dict[str, Any]:
+    """Double button press"""
+    if device_type in ["KS1", "KS1BP"]:
+        # double button click 1-4
+        button = xobj[0]
+        if button == 1:
+            # left button
+            device.fire_event(
+                key=f"{str(EventDeviceKeys.BUTTON)}_left",
+                event_type="double_press",
+                event_properties=None,
+            )
+        if button == 2:
+            # mid left button
+            device.fire_event(
+                key=f"{str(EventDeviceKeys.BUTTON)}_mid_left",
+                event_type="double_press",
+                event_properties=None,
+            )
+        if button == 3:
+            # mid right button
+            device.fire_event(
+                key=f"{str(EventDeviceKeys.BUTTON)}_mid_right",
+                event_type="double_press",
+                event_properties=None,
+            )
+        if button == 4:
+            # right button
+            device.fire_event(
+                key=f"{str(EventDeviceKeys.BUTTON)}_right",
+                event_type="double_press",
+                event_properties=None,
+            )
+        else:
+            return {}
+    return {}
+
+
+def obj560e(
+    xobj: bytes, device: XiaomiBluetoothDeviceData, device_type: str
+) -> dict[str, Any]:
+    """Long button press"""
+    if device_type in ["KS1", "KS1BP"]:
+        button = xobj[0]
+        if button == 1:
```

Comment:
could be simplified to use a mapping
```
BUTTON_TO_NAME = {
  1: "left",
  2: "mid_left",
...
}
```

then `if button_name := BUTTON_TO_NAME(button):`
 ...
```
           device.fire_event(
                key=f"{str(EventDeviceKeys.BUTTON)}_{button_name}",
                event_type="long_press",
                event_properties=None,
            )
```

Reply:
Thanks, will change it tonight,

Reply:
Done